src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PrivateWorkerCraftingRequestResolver.java

```diff
@@ -0,0 +1,167 @@
+package com.minecolonies.coremod.colony.requestsystem.resolvers;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.manager.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requestable.Stack;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.crafting.IRecipeStorage;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.TranslationConstants;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentTranslation;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.RSConstants.CONST_CRAFTING_RESOLVER_PRIORITY;
+
+/**
+ * A crafting resolver which takes care of 2x2 crafts which are crafted by the requesting worker.
+ */
+public class PrivateWorkerCraftingRequestResolver extends AbstractRequestResolver<Stack>
+{
+    public PrivateWorkerCraftingRequestResolver(@NotNull final ILocation location, @NotNull final IToken<?> token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public TypeToken<? extends Stack> getRequestType()
+    {
+        return TypeToken.of(Stack.class);
+    }
+
+    @Override
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<? extends Stack> requestToCheck)
+    {
+        if (!manager.getColony().getWorld().isRemote)
+        {
+            final Colony colony = (Colony) manager.getColony();
+            final ILocation requesterLocation = requestToCheck.getRequester().getRequesterLocation();
+            final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+            if(building instanceof AbstractBuildingWorker)
+            {
+                final ItemStack stack = requestToCheck.getRequest().getStack();
+                if (ItemStackUtils.isEmpty(stack))
+                {
+                    return false;
+                }
+                return ((AbstractBuildingWorker) building).getFirstRecipe(stack) != null;
+            }
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken<?>> attemptResolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends Stack> request)
+    {
+        if (manager.getColony().getWorld().isRemote)
+        {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)